Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -462,25 +462,55 @@ void BlockFormattingContext::resolve_used_height_if_treated_as_auto(Box const& b
         height = max(height, calculate_inner_height(box, available_space.height, computed_values.min_height()));
     }
 
+    auto writing_mode = computed_values.writing_mode();
+    bool has_horizontal_writing_mode = (writing_mode == CSS::WritingMode::HorizontalTb);
+
     if (box.document().in_quirks_mode()
         && box.dom_node()
         && box.dom_node()->is_html_html_element()
-        && box.computed_values().height().is_auto()) {
+        && ((computed_values.width().is_auto() && !has_horizontal_writing_mode)
+            || (computed_values.height().is_auto() && has_horizontal_writing_mode))) {
         // 3.6. The html element fills the viewport quirk
         // https://quirks.spec.whatwg.org/#the-html-element-fills-the-viewport-quirk
-        // FIXME: Handle vertical writing mode.
 
         // 1. Let margins be sum of the used values of the margin-left and margin-right properties of element
         //    if element has a vertical writing mode, otherwise let margins be the sum of the used values of
         //    the margin-top and margin-bottom properties of element.
-        auto margins = box_state.margin_top + box_state.margin_bottom;
+
+        auto writing_mode_dependent_margins = [&]() {
+            if (!has_horizontal_writing_mode)
+                return box_state.margin_left + box_state.margin_right;
+
+            return box_state.margin_top + box_state.margin_bottom;
+        };
+
+        auto margins = writing_mode_dependent_margins();
```

Comment:
I didn't notice that during previous review, but it seems like unless we also update compute_width() to support writing mode, margins will be unresolved by the time we get here.